### PR TITLE
Add external sources section to configuration file

### PR DESCRIPTION
**Pull request by yhagvi**
External sources QL and ext_QL can be specified in confiuration file.

Removed input mode from configuration file.

When synce4l needs to choose the best source, it will consider the local external sources QL it has on the board.

### CODE REVIEW

File: synce_dev.c [hunk 1]
```diff
@@ -208,47 +231,63 @@ static int port_set_ql_line_input(struct synce_dev *dev,
...
+				pr_debug("act on EEC_HOLDOVER for %s",
+					 synce_port_get_name(p));
+				ret = port_set_dnu(p, dev->extended_tlv);
+				if (ret) {
+					pr_err("%s set DNU failed on %s",
+					       __func__, dev->name);
+					return ret;
+				}
+			} else if ((dev->d_state == EEC_LOCKED ||
+				   dev->d_state == EEC_LOCKED_HO_ACQ) &&
```
Comment (kubalewski) [inline synce_dev.c:253]: dev should start in the column as dev in previous line

File: synce_dev.c [hunk 2]
```diff
@@ -278,55 +337,71 @@ static void force_all_eecs_detach(struct synce_dev *dev)
...
 			}
 		}
 	}
 
-	if (best_p) {
-		if (synce_port_is_rx_dnu(best_p)) {
+	if (best_c && best_c->type == PORT) {
+		if (synce_port_is_rx_dnu(best_c->port))
 			return NULL;
+	} else if (best_c && best_c->type == EXT_SRC)
```
Comment (kubalewski) [inline synce_dev.c:394]: while if is covered within {} we should do the same for else if

**Review by kubalewski — CHANGES_REQUESTED** [review]

File: synce_dev.c (hunk 1 above)
Comment (yhagvi) [reply synce_dev.c:253]: fixed

File: synce_dev.c (hunk 2 above)
Comment (yhagvi) [reply synce_dev.c:394]: fixed

File: configs/synce4l.cfg [hunk 3]
```diff
@@ -14,52 +14,24 @@ message_tag		[synce4l]
...
-[<synce1>]
-
-#
-# Mode for synce4l operation.
-# There are two currently supported modes:
-# - line
-# - external
 #
-# Input mode "line" mean the inputs recovered from the PHY's.
+# Clock sources for synchronization can be eigther external clock sources on
```
[the remark this answers is not in the document]
Comment (yhagvi) [reply configs/synce4l.cfg]: fixed

File: synce_clock_source.h [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+	union {
+		struct synce_ext_src *ext_src;
+		struct synce_port *port;
+	};
+};
+
+/**
+ * Alloc memory for a single synce_clock_source instance.
+ *
+ * @param clock_source_name	Human readable name of an clock_sourcel source
```
[the remark this answers is not in the document]
Comment (yhagvi) [reply synce_clock_source.h]: removed the param

File: synce_clock_source.h [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+ *
+ * @param clock_source		Pointer to the clock_source being released
+ */
+void synce_clock_source_destroy(struct synce_clock_source *clock_source);
+
+/**
+ * Compare left with right clock_source, which has higher incoming Quality Level.
+ *
+ * @param left		clock_source instance for comparison
+ * @param righ		clock_source instance for comparison
```
[the remark this answers is not in the document]
Comment (yhagvi) [reply synce_clock_source.h]: fixed

**Review by kubalewski — APPROVED** [review]

**Pull request merged**